### ISSUE TITLE
Error validating model even though it is valid

### ISSUE THREAD

**Issue by jennysharps** [alexandercerutti/passkit-generator#4]
I am getting the following error:

"Provided model ( <model name>  ) matched but unitialized or may not contain icon. Refer to https://apple.co/2IhJr0Q, https://apple.co/2Nvshvn and documentation to fill the model correctly."

After investigating a bit, it seems that the logic in pass.js is as follows:

```
if (!noDynList.length || !noDynList.some(f => f.toLowerCase().includes("icon"))
    || !remoteFilesList.some(f => f.toLowerCase().includes("icon"))) {
    let eMessage = formatError("MODEL_UNINITIALIZED", path.parse(this.model).name);
    throw new Error(eMessage);
}
```

This requires me to have an icon both in the noDynList AND the remoteFilesList; surely we only need 1 or the other rather than both?

When I change the code to this it works as I would expect, and I can open the genereted pass on my phone:
```
if (!(noDynList.length || noDynList.some(f => f.toLowerCase().includes("icon"))
    || remoteFilesList.some(f => f.toLowerCase().includes("icon")))) {
    let eMessage = formatError("MODEL_UNINITIALIZED", path.parse(this.model).name);
    throw new Error(eMessage);
}
```

Is my understanding of the expected behavior here wrong? Is the icon actually needed in both lists for some reason?

**Comment by alexandercerutti** [comment]
No, the icon is needed at least in one. If it is present both in noDynList or remoteFilesList, the one provided in remoteFilesList will be chosen.

Your solution is interesting, but it misses `noDynList` if it has some files in it check (`noDynList.length`). Anyway I keep don't understanding the reason of the error. The original check should be valid.

Moreover, I notice another thing: the message error does not contain the model name in the parenthesis as expected (if you did not remove it).

I should investigate more...

P.s. I added syntax highlighting to your code :)

**Comment by jennysharps** [comment]
I did remove the model name, sorry -- didn't think it was relevant, but it did show the correct model name in the error.

Basically, when I logged the noDynList I had 1 icon file, and when I logged the remoteFilesList I had none, so this was failing because since the first 2 checks evaluated to false, it was falling back to the third check:
1. !noDynList.length evaluated false because noDynList had files
2. !noDynList.some(f => f.toLowerCase().includes("icon")) evaluated to false because noDynList did have a matching icon file
3. !remoteFilesList.some(f => f.toLowerCase().includes("icon")) evaluated to true because remoteFilesList was an empty array

Therefore if (false || false || true) evaluated to true, so threw an error.  

Why is the noDynList length check needed?  Is it because that list must have at least 1 file in order to be valid? If so then would this work?

```javascript```
if (!noDynList.length 
 || ![...noDynList, ...remoteFilesList].some(f => f.toLowerCase().includes("icon"))) {
    let eMessage = formatError("MODEL_UNINITIALIZED", path.parse(this.model).name);
    throw new Error(eMessage);
}
```

Thanks for your response by the way!

Thanks for adding syntax highlighting. :)

**Comment by alexandercerutti** [comment]
You are right.

A check on noDynList is needed because if you use an empty model, your manifest will be empty, the archive will be empty too and you would generate an invalid pass. The same happens if the pass does not have an icon: the pass is invalid.

I think it may work and also you merge the icon check logic, so it is faster.
What about making a pull request referencing this issue? 😊

**Comment by jennysharps** [comment]
👍Great, will do as soon as I have a chance! I'm going to raise another issue soon too since I'm actually creating an NFC-enabled pass & the code doesn't quite work as-is but will focus on this one first!

**Comment by alexandercerutti** [comment]
Ops, I misunderstood, but still prefer the first way. 😊

Awesome! If you want to add something to the documentation about NFC (something I did not added yet), feel free to do it.

Well, okay! I'll wait for your issue post. 🤙

**Comment by alexandercerutti** [comment]
Hi, you have to fork the repo with the button above, near Star/Unstar. You'll get the latest version of the repo available on GitHub. Then you'll have to apply changes on the new repo, through cloning or via Github Web UI.

Once done, you'll have to go to Pull Requests > `New Pull Request`.
You'll get this interface. Click on `compare across forks` and select yours from the right couple of dropdowns, the left one.

<img width="999" alt="image" src="https://user-images.githubusercontent.com/5442545/49510970-1bc6f880-f88a-11e8-8f3f-ce4d572c5349.png">

😊

**Comment by jennysharps** [comment]
Thanks!
